packages/core/src/index.ts

```diff
@@ -692,7 +693,20 @@ class Hopr extends EventEmitter {
     let addrToAnnounce: Multiaddr
 
     if (includeRouting) {
-      const multiaddrs = await this.getAnnouncedAddresses()
+      let multiaddrs = await this.getAnnouncedAddresses()
+      multiaddrs.sort((a, b) => {
```

Comment:
Why don't drop all private / localhost addresses if `preferLocalAddresses` is not set?

Reply:
We anyway "announce" only one address, so we don't even need an array

Reply:
Seems like a good point!